sklearn/ensemble/forest.py

```diff
@@ -242,6 +242,12 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
+
+        if self.n_estimators == 'warn':
```

Comment:
The check and validation should be done in `fit` instead of `__init__`

Reply:
So should I change back to n_estimators=10 instead of n_estimators='warn', and then change my if conditional check in the fit() method?

Reply:
no the warn is good, just the test should be in the other place.

Reply:
You can refer to: https://github.com/scikit-learn/scikit-learn/pull/11469/files#diff-e6faf37b13574bc591afbf0536128735R864

This is still not merged but we follow this convention: `__init__` just assign the parameters to the class attributes and we do checking and validation in the `fit` method.

Reply:
Aren't lines 245 and 246 above inside the fit() method?

Reply:
Ups sorry it is good there. I good confused with another PR :)